src/main/kotlin/graphics/scenery/CurveGeometry.kt

```diff
@@ -0,0 +1,189 @@
+ */
+class CurveGeometry(curve: CatmullRomSpline, n: Int = 100): Node("CurveGeometry"), HasGeometry {
+    override val vertexSize = 3
+    override val texcoordSize = 2
+    override var geometryType = GeometryType.TRIANGLES
+
+    override var vertices: FloatBuffer = BufferUtils.allocateFloat(0)
+    override var normals: FloatBuffer = BufferUtils.allocateFloat(0)
+    override var texcoords: FloatBuffer = BufferUtils.allocateFloat(0)
+    override var indices: IntBuffer = BufferUtils.allocateInt(0)
+
+    private val cur = curve.CatMulRomChain(n)
+
+    /**
+     * This function renders the spline.
+     * @param [baseShape] It takes a lambda as a parameter, which is the shape of the
+     * curve. If you choose, for example, to have a square as a base shape, your spline will look like
+     * a banister. Please not that the base shape needs an equal number of points in each segments but it
+     * can very well vary in thickness.
+     */
+    fun drawSpline(baseShape: (() -> List<GLVector>)){
+        data class TranslationMatrix(val matrix: GLMatrix, val translation: GLVector)
+        val bases = ArrayList<TranslationMatrix>()
+        computeFrenetFrames().forEach { (t, n, b, tr) ->
+            if(n != null && b != null) {
+                val basisArray = ArrayList<Float>()
+                basisArray.add(n.x())
+                basisArray.add(b.x())
+                basisArray.add(t.x())
+                basisArray.add(0f)
+                basisArray.add(n.y())
+                basisArray.add(b.y())
+                basisArray.add(t.y())
+                basisArray.add(0f)
+                basisArray.add(n.z())
+                basisArray.add(b.z())
+                basisArray.add(t.z())
+                basisArray.add(0f)
+                basisArray.add(0f)
+                basisArray.add(0f)
+                basisArray.add(0f)
+                basisArray.add(1f)
+                val array = basisArray.toFloatArray()
+                val matrix = GLMatrix(array)
+                val translationMatrix = TranslationMatrix(matrix, tr)
+                bases.add(translationMatrix)
+            }
+        }
+
+        val curveGeometry = ArrayList<ArrayList<GLVector>>()
+        bases.forEach { (m,t) ->
+            val basis = m.inverse
+            val shape = ArrayList<GLVector>()
+            baseShape.invoke().forEach {
+                val vector4D = GLVector(it.x(), it.y(), it.z(), 0f)
+                val vector = basis.mult(vector4D)
+                val vertex = GLVector(vector.x()+t.x(), vector.y()+t.y(), vector.z()+t.z())
+                shape.add(vertex)
+            }
+            curveGeometry.add(shape)
+        }
+
+        val verticesVectors = ArrayList<GLVector>()
+
+        curveGeometry.forEachIndexed { shapeIndex, shape ->
+            if (shapeIndex < curveGeometry.lastIndex) {
+                shape.forEachIndexed { vertexIndex, vertex ->
+                    if(vertexIndex < shape.lastIndex) {
```

Comment:
Instead of having an if for each of the indices, consider doing sth like `shape.dropLast(1).forEachIndexed`